### PR TITLE
angular 9 upgrade

### LINKED ISSUE #424: Angular 9 upgrade
Upgrade metrics application to Angular 9

### PR DESCRIPTION
**Pull request by sandikbarr**
closes #424

### CODE REVIEW

File: src/app/core/routing/routing.service.spec.ts [hunk 1]
```diff
@@ -63,16 +64,16 @@ describe('RoutingService', () => {
     })
       .compileComponents()
       .then(() => {
-        routingService = TestBed.get(RoutingService);
-        store = TestBed.get(Store);
-        router = TestBed.get(Router);
+        routingService = TestBed.inject(RoutingService);
```
Comment (sandikbarr) [inline src/app/core/routing/routing.service.spec.ts:67]: TestBed.get has been deprecated in favor of typed TestBed.inject https://angular.io/api/core/testing/TestBed#get

File: src/app/ngrx/effects/castle-catalog.effects.spec.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import { Actions } from '@ngrx/effects';
 import { TestBed, async } from '@angular/core/testing';
 import { cold, hot } from 'jasmine-marbles';
-import { StoreModule, Store } from '@ngrx/store';
+import { StoreModule, Store, Action } from '@ngrx/store';
 import { EffectsModule } from '@ngrx/effects';
-import { getActions, TestActions } from './test.actions';
-
+import { provideMockActions } from '@ngrx/effects/testing';
```
Comment (sandikbarr) [inline src/app/ngrx/effects/castle-catalog.effects.spec.ts:6]: ngrx adds `provideMockActions` for testing effects https://ngrx.io/guide/effects/testing

File: src/app/ngrx/effects/castle-downloads.effects.ts [hunk 3]
```diff
@@ -1,298 +1,287 @@
...
-          interval: this.routerParams.interval,
-          beginDate: this.routerParams.beginDate,
-          endDate: this.routerParams.endDate
-        });
-      });
-    })
+      }),
+      mergeMap(([action, routerParams]) => {
+        const { episodes } = action;
+        return [
```
Comment (sandikbarr) [inline src/app/ngrx/effects/castle-downloads.effects.ts:34]: prefer returning array of `Action`s rather than dispatching from effects

**Review by rpeterman-gp — APPROVED** [review]
Looks good to me. Nice Work!

**Pull request merged**